cms/djangoapps/contentstore/views/transcript_settings.py

```diff
@@ -108,3 +119,138 @@ def transcript_credentials_handler(request, course_key_string):
+    must_have_attrs = ['edx_video_id', 'language_code', 'new_language_code']
+    missing = [attr for attr in must_have_attrs if attr not in data]
+    if missing:
+        error = _(u'Following parameters are required: {missing}.').format(missing=', '.join(missing))
+    elif (
+        data['language_code'] != data['new_language_code'] and
+        data['new_language_code'] in get_available_transcript_languages([data['edx_video_id']])
+    ):
+        error = _(u'Transcript with "{language_code}" language code is already present.'.format(
+            language_code=data['new_language_code']
+        ))
+    elif 'file' not in files:
+        error = _(u'A transcript file is required.')
+
+    return error
+
+
+@login_required
+@require_POST
+def transcript_upload_handler(request, course_key_string):
+    """
+    View to upload a transcript file.
+
+    Arguments:
+        request: A WSGI request object
+        course_key_string: Course key identifying a course
+
+    Transcript file, edx video id and transcript language are required.
+    Transcript file should be in SRT(SubRip) format.
+
+    Returns
+        - A 400 if any of the validation fails
+        - A 404 if the corresponding feature flag is disabled
+        - A 200 if transcript has been uploaded successfully
+    """
+    # Check whether the feature is available for this course.
+    course_key = CourseKey.from_string(course_key_string)
+    if not VideoTranscriptEnabledFlag.feature_enabled(course_key):
+        return HttpResponseNotFound()
+
+    error = validate_transcript_upload_data(data=request.POST, files=request.FILES)
+    if error:
+        response = JsonResponse({'error': error}, status=400)
+    else:
+        edx_video_id = request.POST['edx_video_id']
+        language_code = request.POST['language_code']
+        new_language_code = request.POST['new_language_code']
+        transcript_file = request.FILES['file']
+        try:
+            # Convert SRT transcript into an SJSON format
+            # and upload it to S3.
+            sjson_subs = Transcript.convert(
+                content=transcript_file.read(),
+                input_format=Transcript.SRT,
+                output_format=Transcript.SJSON
+            )
+            create_or_update_video_transcript(
+                video_id=edx_video_id,
+                language_code=language_code,
+                metadata={
+                    'provider': TranscriptProvider.CUSTOM,
+                    'file_format': Transcript.SJSON,
+                    'language_code': new_language_code
+                },
+                file_data=ContentFile(json.dumps(sjson_subs)),
+            )
+            response = JsonResponse(status=201)
+        except (TranscriptsGenerationException, UnicodeDecodeError):
+            response = JsonResponse(
+                {'error': _(u'There is a problem with this transcript file. Try to upload a different file.')},
```

Comment:
Sylvia can you look at this string - This is mostly like not user-facing string?